cmd/ef_tests/state/runner_v2/report.rs

```diff
@@ -0,0 +1,107 @@
+        .open(failing_report_path)
+        .unwrap();
+    let content = format!(
+        "Test checks failed for test: {:?}. \nTest path: {:?}\nTest description/comment: {}\nTest doc reference: {}\n ",
+        test.name,
+        test.path,
+        test._info.description.clone().unwrap_or(
+            test._info
+                .comment
+                .clone()
+                .unwrap_or("This test has no description or comment".to_string())
+        ),
+        test._info
+            .reference_spec
+            .clone()
+            .unwrap_or("This test has no reference spec".to_string())
+    );
+    report.write_all(content.as_bytes()).unwrap();
+
+    for check_result in failing_test_cases {
+        let content = format!("\n{}", check_result);
+        report.write_all(content.as_bytes()).unwrap();
+    }
+    let dividing_line = "-----------------------------------------------------\n\n".to_string();
+    let _ = report.write_all(dividing_line.as_bytes());
+}
+
+impl fmt::Display for PostCheckResult {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        writeln!(f, "Fork: {:?} - vector {:?}\n", self.fork, self.vector)?;
+        if let Some(root_mismatch) = self.root_dif {
+            writeln!(
+                f,
+                "  ERR - ROOT MISMATCH:\n    Expected root: {:?}\n    Actual   root: {:?}",
+                root_mismatch.0, root_mismatch.1
+            )?;
+        }
+
+        if let Some(exception_diff) = self.exception_diff.clone() {
+            writeln!(
+                f,
+                "  ERR - EXCEPTION MISMATCH:\n    Expected exception: {:?}\n    Actual   exception: {:?}",
+                exception_diff.0, exception_diff.1
+            )?;
+        }
+
+        if let Some(logs_mismatch) = self.logs_diff {
+            writeln!(
+                f,
+                "  ERR - LOGS MISMATCH:\n    Expected logs hash: {:?}\n    Actual   logs hash: {:?}",
+                logs_mismatch.0, logs_mismatch.1
+            )?;
+        }
+
+        if let Some(account_mismatches) = self.accounts_diff.clone() {
+            for acc_mismatch in account_mismatches {
+                writeln!(
+                    f,
+                    "  ERR - ACCOUNT STATE MISMATCH:\n    Address: {:?}\n     Expected balance: {:?}\n     Actual   balance: {:?}\n     Expected nonce: {:?}\n     Actual   nonce: {:?}\n     Expected code: 0x{}\n     Actual   code: 0x{}\n     Expected storage: {:?}\n     Actual   storage: {:?}",
+                    acc_mismatch.address,
+                    acc_mismatch.expected_balance,
+                    acc_mismatch.actual_balance,
+                    acc_mismatch.expected_nonce,
+                    acc_mismatch.actual_nonce,
+                    hex::encode(acc_mismatch.expected_code),
+                    hex::encode(acc_mismatch.actual_code),
+                    acc_mismatch.expected_storage,
+                    acc_mismatch.actual_storage
+                )?;
```

Comment:
Showing all fields can get a bit spammy, I think it would be more useful to only show the fields that don't match (ie if both accounts have the same balance, we don't need to show the balance). We could achieve this by implementing `Display` for `AccountMismatch`

Reply:
Also, the `code` field can get very long to show in the report, maybe we can show its hash instead?